src/core/qgsunittypes.h

```diff
@@ -563,6 +563,13 @@ class CORE_EXPORT QgsUnitTypes
      */
     Q_INVOKABLE static QString formatAngle( double angle, int decimals, QgsUnitTypes::AngleUnit unit );
 
+    /**
+     * Returns an angle formatted as a friendly string.
+     * \param angle angle to format
+     * \param unit unit of angle
+     * \returns formatted angle string
+     */
+    Q_INVOKABLE static QString formatAngle( double angle, QgsUnitTypes::AngleUnit unit );
```

Comment:
Could you add a "decimals" argument here, default to -1 and add a note that "-1 means automatically pick an appropriate number of decimal places for the unit". (also needs some unit tests)

Reply:
Thank you I'll implement the "-1" suggestion. There is already an overload which takes a decimals [parameter](https://github.com/qgis/QGIS/blob/83a441589a544015897640f6a8bd06043f59f9d6/src/core/qgsunittypes.h#L564), but it is the second param so I can't set the default value to -1. Or should I set a default parameter for the unit to? like `=AngleDegrees`?

Reply:
I have removed this function and modified the original one which now can take -1 for decimals. Unit tests where updated too.